Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
 name = "firestore-db-and-auth"
-version = "0.6.1"
+version = "0.8.0"
```

Comment:
I would argue this should probably be a major version breaking change. With the changes proposed here all `sync` code disappears - meaning this library becomes exclusively `async`.  Changes here would no longer be compatible with those using the sync only version of this library